### ISSUE TITLE
autohide: leave requires double tap on touch devices

### ISSUE THREAD

**Issue by itshallrun** [KingSora/OverlayScrollbars#285]
Hey,

I am working on a cross plattform web app.
The issue is, on touch devices, the autoHide: "leave"  option requires a double tap to trigger the touch event.
The first tap will show the scrollbar, and the second tap will actually trigger the touch event underneath.

touchSupport  is active in my options.

```
                    resize: "none",
                    sizeAutoCapable: true,
                    paddingAbsolute: true,
                    scrollbars: {
                        visibility: "auto",
                        autoHide: "leave",
                        autoHideDelay: 400,
                        dragScrolling: true,
                        clickScrolling: false,
                        touchSupport: true,
                    },
````

Is there any way to have autoHide: "leave" for non-mobile devices and trigger the touch event on the first tap on mobile?

**Comment by KingSora** [comment]
Good day!

With " trigger the touch event on the first tap" you mean the possibility to scroll, or the possibility to interact with the scrollbars?

**Comment by itshallrun** [comment]
I mean, if there are some elements in a scrollable container, then the first tap will show the scrollbars (which are hidden by default) and the second tap will trigger the onClick / touch event of the touched element.
So the scrollbars are kind of blocking the content from being triggered on the first tap.

**Comment by KingSora** [comment]
Alright.. The touched elements are they below the actual scrollbars, or does the position of the element doesn't matter?

What OS, device and browser are you using? Does this only happen with the `autohide: leave` option? Can you trigger it also on desktop devices somehow?

**Comment by itshallrun** [comment]
I am using the react version.
I could test it on an Ipad pro. 
It is an issue for `autohide:leave` only.
The position doesn't matter. 
Whenever I tap in the Scrollbars container, the first tap always triggers the visibility of the scrollbars.
On the desktop, the behavior is as desired. The scrollbars are hidden until the scrollbars container is hovered. And they disappear when the container is left.

**Comment by KingSora** [comment]
@itshallrun alright, I'll take a look into it as soon as I have time.. I don't have a iPad, so I'll need some time to find a way to reproduce it

**Comment by beeteam368** [comment]
Hello KingSora,

Thanks for great library!

I tested it very thoroughly on the iPhone.

I confirm that this is a problem when using on iOS.

Please review and fix it when possible!

Best Regards!

**Comment by KingSora** [comment]
@beeteam368 I'll give my best :)

**Comment by beeteam368** [comment]
> @beeteam368 I'll give my best :)

Thanks! Wish you a good day!

**Comment by KingSora** [comment]
Can you please try `v2` and confirm whether this issue still persists? :)

**Comment by alvarolb** [comment]
Yes, I confirm that the issue is still present on 2.1.0.

The first touch displays the current scroll, and the second lets you click on the link.

**Comment by robwheatley** [comment]
Hi there.
First off @KingSora - loving your work!

Just to add to this - I've also just hit the same problem and came across this page looking for a solution. My guess is that it's down to the fact that iOS doesn't like hovers. Anything with a hover effect on it needs a double tap in iOS, and I assume you have a hover somewhere looking for the mouse hover on auto hide.

Anyway - I hacked a work-around to get me going - at least for now. No idea if it's any good, but it only took 2 mins and it seems to work.

I installed[ react-device-detect](https://github.com/duskload/react-device-detect) (something I have used before) and then created a new component that uses the fact that you are on a mobile device to skip using OverlayScrollbars. The scrollbars do what I want them to do on iOS out of the box anyway.

Kinda like this:

```
import { isMobile } from 'react-device-detect';

function ScrollWrapper({ children }) {

  if (isMobile) {
    return children
  } else {
    return (
      <OverlayScrollbarsComponent
        options={{ scrollbars: { autoHide: 'leave', autoHideDelay: 500 } }}
      >
        {children}
      </OverlayScrollbarsComponent>
    )
  }
}
```

Once you have the above, replace where you had the OverlayScrollbarsComponent with ScrollWrapper in your original code.

I hope that makes sense.

**Comment by KingSora** [comment]
@robwheatley Thanks for posting your solution :)

Just for completeness I would like to add that this solution could cause problems with server side rendering

**Comment by robwheatley** [comment]
@KingSora Thanks for the heads-up. In my case, I don't need to worry as I'm not doing any SSR ;-). Worth noting though.

**Comment by KingSora** [comment]
This issue should be fixed in version [`v2.3.1`](https://www.npmjs.com/package/overlayscrollbars/v/2.3.1).
